### ISSUE TITLE
static-marks returns no error when it failed to parse the yml-file

### ISSUE THREAD

**Issue by gaerfield** [darekkay/static-marks#10]
Given the following bookmarks-yaml-file:
```yaml
test:
  - duckduckgo: https://duckduckgo.com/
    - more duck: https://duckduckgo.com/
```
static-marks will output an error (which is absolutely fine), but the return value is still '0':
![image](https://user-images.githubusercontent.com/65171481/84890124-45c1c380-b09a-11ea-8916-8d0b1cfe1903.png)

Further information: Within an automatic build, this will result in a html-file containing no links whereas the build just runs fine. This finally results in overwriting a fine version of a page with an empty one.

**Comment by darekkay** [comment]
Thanks for the report!

My reasoning behind this: when multiple files are being used, one malformed file should not prevent the tool from building all other (valid) files. But I am not sure about this decision anymore. I think a stricter handling might indeed outweigh the pros of trying to have a [graceful degradation](https://searchnetworking.techtarget.com/definition/graceful-degradation).

The following flow should address the issues:

1. Keep the current output for every malformed file.
2. Abort the build command if at least one file is malformed. This includes not overwriting an existing output file (both with `-o` and `>`).
3. Exit with code `1`.

**Comment by gaerfield** [comment]
If you fear this to be a breaking change, maybe introduce an option "--strict"? This would also fine be me.

**Comment by darekkay** [comment]
Implemented in [2.2.0](https://github.com/darekkay/static-marks/releases/tag/v2.2.0).

**Comment by gaerfield** [comment]
![grafik](https://user-images.githubusercontent.com/13051868/85977854-b099c580-b9dd-11ea-8d86-ae7b2bc3017f.png)
awesome
